src/fuzz-tokenize.zig

```diff
@@ -55,17 +55,90 @@ pub fn zig_fuzz_test_inner(buf: [*]u8, len: isize, debug: bool) void {
+
+    // Second tokenization.
+    tokenizer = tokenize.Tokenizer.init(buf2.items, &messages, &gcd, gpa);
+    tokenizer.tokenize();
+    var output2 = tokenizer.finish_and_deinit();
+    defer output2.tokens.deinit(gpa);
+
+    if (debug) {
+        std.debug.print("After:\n", .{});
+        for (0..output2.tokens.tokens.len) |token_index| {
+            const token = output2.tokens.tokens.get(token_index);
+            std.debug.print("\t{any}\n", .{token});
+        }
+        std.debug.print("\n\n", .{});
+    }
+    // Assert same.
+    var same = output.tokens.tokens.len == output2.tokens.tokens.len;
+    for (0..output.tokens.tokens.len) |token_index| {
+        if (token_index >= output2.tokens.tokens.len) {
+            same = false;
+            break;
+        }
+        const token = output.tokens.tokens.get(token_index);
+        const token2 = output2.tokens.tokens.get(token_index);
+        same = same and (token.tag == token2.tag);
+        same = same and (token.offset == token2.offset);
+        same = same and (token.length == token2.length);
+    }
+
+    if (!same) {
+        var prefix_len: usize = 0;
+        var suffix_len: usize = 0;
+
+        while (prefix_len < output.tokens.tokens.len and prefix_len < output2.tokens.tokens.len) : (prefix_len += 1) {
+            const token = output.tokens.tokens.get(prefix_len);
+            const token2 = output2.tokens.tokens.get(prefix_len);
+            if (token.tag != token2.tag or token.offset != token2.offset or token.length != token2.length) {
+                break;
+            }
+        }
+
+        while (suffix_len < output.tokens.tokens.len - prefix_len and suffix_len < output2.tokens.tokens.len - prefix_len) : (suffix_len += 1) {
+            const token = output.tokens.tokens.get(output.tokens.tokens.len - suffix_len - 1);
+            const token2 = output2.tokens.tokens.get(output2.tokens.tokens.len - suffix_len - 1);
+            if (token.tag != token2.tag or token.offset != token2.offset or token.length != token2.length) {
+                break;
+            }
+        }
+
+        std.debug.print("...\n", .{});
+        for (prefix_len..output.tokens.tokens.len - suffix_len) |token_index| {
+            const token = output.tokens.tokens.get(token_index);
+            std.debug.print("\x1b[31m\t- {any}\x1b[0m: {s}\n", .{ token, buf_slice[token.offset..][0..token.length] });
+        }
+        for (prefix_len..output2.tokens.tokens.len - suffix_len) |token_index| {
+            const token = output2.tokens.tokens.get(token_index);
+            std.debug.print("\x1b[32m\t+ {any}\x1b[0m: {s}\n", .{ token, buf2.items[token.offset..][0..token.length] });
+        }
+        std.debug.print("...\n", .{});
+
+        std.debug.assert(same);
+    }
+}
+
+fn rebuild_buffer(buf: []const u8, tokens: *tokenize.Token.List, alloc: std.mem.Allocator) !?std.ArrayList(u8) {
+    // Create an arraylist to store the new buffer.
+    var buf2 = std.ArrayList(u8).init(alloc);
```

Comment:
Oh also, please allocate this with the same capacity as `buf`. That will definitely make this run faster cause it should never need to reallocate.